Merge meta data in setup.cfg into pyproject.toml

KyleQianliMa opened the pull request:
# Short description of the changes:
<!-- Add a concise description here-->
Merging meta data in setup.cfg into pyproject.toml.
This is to address [Story 3163](https://ornlrse.clm.ibmcloud.com/ccm/web/projects/Neutron%20Data%20Project%20(Change%20Management)#action=com.ibm.team.workitem.viewWorkItem&id=3163)
# Long description of the changes:
<!-- Optional, add more details here if the short description is not suffieicnt-->

# Check list for the pull request
- [ ] I have read the [CONTRIBUTING]
- [ ] I have read the [CODE_OF_CONDUCT]
- [ ] I have added tests for my changes
- [ ] I have updated the documentation accordingly

# Check list for the reviewer
- [ ] I have read the [CONTRIBUTING]
- [ ] I have verified the proposed changes
- [ ] best software practices
    + [ ] all internal functions have an underbar, as is python standard
    + [ ] clearly named variables (better to be verbose in variable names)
    + [ ] code comments explaining the intent of code blocks
- [ ] All the tests are passing
- [ ] The documentation is up to date
- [ ] code comments added when explaining intent

# Manual test for the reviewer
<!-- Instructions for testing here. -->

# References
<!-- Links to related issues or pull requests -->
<!-- Links to IBM EWM items if aaplicable -->

jmborr commented on pyproject.toml:
```diff
@@ -1,22 +1,48 @@
+]
 description = "Reduction scripts for the Liquids Reflectometer. This includes both automated reduction scripts and useful scripts to reprocess data."
-dynamic = ["version", "readme", "license", "dependencies"]
+dynamic = ["version"]
```
```suggestion
version = "0.1.0"  # initial version, dynamically overriden
```

jmborr commented on pyproject.toml:
```diff
@@ -1,22 +1,48 @@
-name = "LiquidsReflectometer"
+name = "lr_reduction"
+authors = [
+	{name = "Mat", email = "m2d@ornl.gov"}
```
change tab to spaces

jmborr commented on pyproject.toml:
```diff
@@ -1,22 +1,48 @@
+    "numpy",
+    "matplotlib",
+    "mantid"
+
```
remove the empty line

jmborr commented on pyproject.toml:
```diff
@@ -1,22 +1,48 @@
 classifiers = [
   "Operating System :: OS Independent",
   "Programming Language :: Python :: 3"
 ]
```
"Topic :: Scientific/Engineering :: Neutron reflectometry"

jmborr commented on pyproject.toml:
```diff
@@ -1,22 +1,48 @@
+    "mock",
+    "pytest",
+    "pytest-qt",
+    "typing"
```
remove `typing`, `mock`, and `pytest-qt`

jmborr reviewed (changes requested):
did you try installing the package with `pip install -e .`?

KyleQianliMa commented:
added 
[tool.setuptools.packages.find]
where = ["reduction"]
exclude = ["tests*", "scripts*", "docs*", "notebooks*"] in pyproject.toml and addressed other comments.

KyleQianliMa commented:
I think the conflict has been resolved and now ready to merge

jmborr reviewed (approved):
`pip install -e .` now installs package `lr_reduction` so that later one can do `import lr_reduction`.
